data/scripts.js

```diff
@@ -1172,6 +1172,10 @@ let BattleScripts = {
 			if (!pokemon.side.canDynamax) return;
 			if (this.canZMove(pokemon)) return;
```

Comment:
This does reveal the actual problem, though: Once a Pokémon has used its Z-move or mega evo, it's no longer governed by these checks and can Dynamax.